src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -451,6 +494,41 @@ public void Close()
             }
         }
 
+        /// <summary>
+        /// Write the XDocument to the packages.config and disallow further changes.
+        /// </summary>
+        /// <param name="fullPath">the full path to packages.config file</param>
+        public void Close(string fullPath)
+        {
+            var directorypath = Path.GetDirectoryName(fullPath);
+
+            var tempConfigPath = Path.Combine(directorypath, Path.GetRandomFileName());
+
+            using (var tempConfigStream = File.Create(tempConfigPath))
+            {
+                // Save the XDocument to the temp stream
+                _xDocument.Save(tempConfigStream);
+            }
+
+            // Delete the existing packages.config file
+            if (File.Exists(fullPath))
+            {
+                // Make file deletable by removing read-only attribute
+                // such as under source control
+                var attributes = File.GetAttributes(fullPath);
+
+                if (attributes.HasFlag(FileAttributes.ReadOnly))
+                {
+                    File.SetAttributes(fullPath, attributes & ~FileAttributes.ReadOnly);
+                }
+
+                File.Delete(fullPath);
+            }
+
+            // Rename the temporary file to packages.config file
+            File.Move(tempConfigPath, fullPath);
```

Comment:
you want to catch and log if something went wrong here, this is a catastrophic failure if it does.